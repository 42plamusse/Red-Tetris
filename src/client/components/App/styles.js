import styled from "styled-components";

export const StyledContainer = styled.div`
  position: fixed;
  width: 100%;
  display: flex;
  flex-direction: column;
`;
