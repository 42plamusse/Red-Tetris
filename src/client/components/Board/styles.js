import styled, { css } from "styled-components";
import { PURE_ORANGE, CRIMSON } from "../../../constants/colors";

export const StyledAspectRatio = styled.div`
  padding-top: 200%;
  width: 100%;
  height: 0;
  position: relative;
  // box-sizing: border-box;
`;

export const StyledAspectRatioContainer = styled.div`
  flex-basis: 25%;
  // box-sizing: border-box;
  text-align: center;
  max-width: 500px;
  min-width: 325px;
  ${props =>
    props.isSpectre &&
    css`
      flex-basis: 30%;
      padding: 1em;
      min-width: 100px;
      max-width: 150px;
    `};
`;

export const StyledContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-left: 0px;
  border: 4px solid black;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  // box-sizing: border-box;
  ${props =>
    props.isSpectre &&
    css`
      border: 2px solid black;
      background: linear-gradient(${PURE_ORANGE} 0%, ${CRIMSON} 100%);
    `};
`;

export const StyledRow = styled.div`
  flex: 1;
  display: flex;
`;
