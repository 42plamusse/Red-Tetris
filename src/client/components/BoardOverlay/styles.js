import styled, { css } from "styled-components";

export const StyledContainer = styled.div`
  align-self: center;
  display: flex;
  justify-content: center;
  border: 3px solid white;
  border-radius: 5px;
  background-color: rgba(0, 0, 0, 0.5);
  position: absolute;
  top: -3px;
  height: 86vh;
  width: 43vh;
`;

export const StyledGameOverText = styled.h1`
  align-self: center;
  font-size: 5vh;
`;
