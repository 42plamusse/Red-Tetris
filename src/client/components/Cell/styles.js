import styled, { css } from "styled-components";

export const StyledContainer = styled.div`
  flex: 1 10%;
  box-sizing: border-box;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
  border: 1px solid black;
  ${props =>
    props.color &&
    css`
      box-sizing: border-box;
      -moz-box-sizing: border-box;
      -webkit-box-sizing: border-box;
      border: 5px outset ${props => props.color};
      background-color: ${props => props.color};
    `};
`;
