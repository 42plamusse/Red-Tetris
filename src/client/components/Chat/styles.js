import styled from "styled-components";

export const StyledMainContainer = styled.div`
  flex: 4;
  padding: 20px;
  margin: 5px;
  background-color: rgba(0, 0, 0, 0.9);
  // border: 5px solid white;
  border-radius: 2px;
`;
