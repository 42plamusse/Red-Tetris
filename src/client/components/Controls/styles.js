import styled, { css } from "styled-components";

export const StyledContainer = styled.div`
  align-self: center;
  display: flex;
  flex-direction: column;
  div {
    display: flex;
    justify-content: space-between;
    margin-top: 20px;
    kbd {
      font-size: 2vh;
      border-radius: 3px;
      padding: 1px 2px 0;
      border: 3px solid white;
    }
    span {
      font-size: 1.5vh;
    }
  }
  h2 {
    margin-top: 40px;
    align-self: center;
  }
`;
