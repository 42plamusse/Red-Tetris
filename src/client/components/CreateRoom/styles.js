const styles = {
  createRoomContainer: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center"
  }
};

export default styles;
