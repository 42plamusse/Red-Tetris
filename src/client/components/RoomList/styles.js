import styled from "styled-components";

export const StyledMainContainer = styled.div`
  display: flex;
  flex: 1;
  margin: 10px;
`;
