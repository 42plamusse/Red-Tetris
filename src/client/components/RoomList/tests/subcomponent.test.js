import React from "react";
import { shallow, render } from "enzyme";
import toJson from "enzyme-to-json";
import RoomListSub from "../subcomponent";
import { StyledListItem } from "../styles";

const props = {
  rooms: [
    { name: "Room1", playersCount: 2, isStarted: true },
    { name: "Room2", playersCount: 3, isStarted: false }
  ],
  playerName: "Player1"
};

describe("<RoomListSub />", () => {
  test("should render without crashing", () => {
    const wrapper = render(<RoomListSub {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("should joinRoom when clicking a roomListItem", () => {
    const wrapper = shallow(<RoomListSub {...props} />);

    const item = wrapper.find(StyledListItem);
    expect(window.location.hash).toBe("");
    item.first().simulate("click");
    expect(window.location.hash).toBe(
      "#" + props.rooms[0].name + "[" + props.playerName + "]"
    );
  });
});
