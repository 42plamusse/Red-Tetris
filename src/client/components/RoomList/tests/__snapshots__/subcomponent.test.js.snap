// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoomListSub /> should render without crashing 1`] = `
<div
  class="sc-bdVaJa mWeMt"
  id="room-list"
>
  <h2>
    Rooms :
  </h2>
  <div
    class="sc-bwzfXH hQcXqj"
    id="list"
  >
    <div
      class="sc-ifAKCX FSPBW"
      id="subheader"
    >
      <div
        class="sc-EHOje kgNXRr"
      >
        <p
          id="header"
        >
          Name
        </p>
      </div>
      <div
        class="sc-EHOje qZCVm"
      >
        <p
          id="header"
        >
          Mode
        </p>
      </div>
      <div
        class="sc-EHOje qZCVm"
      >
        <p
          id="header"
        >
          Players
        </p>
      </div>
      <div
        class="sc-EHOje qZCVm"
      >
        <p
          id="header"
        >
          Started
        </p>
      </div>
    </div>
    <div
      class="sc-htpNat eHEvwL"
      id="list-item"
    >
      <div
        class="sc-bxivhb bKDjvz"
      >
        <p
          id="content"
        >
          Room1
        </p>
      </div>
      <div
        class="sc-bxivhb iuOKpD"
      >
        <p
          id="content"
        >
          BattleRoyal
        </p>
      </div>
      <div
        class="sc-bxivhb iuOKpD"
      >
        <p
          id="content"
        >
          2/10
        </p>
      </div>
      <div
        class="sc-bxivhb iuOKpD"
      >
        <p
          id="content"
        >
          Yes
        </p>
      </div>
    </div>
    <div
      class="sc-htpNat eHEvwL"
      id="list-item"
    >
      <div
        class="sc-bxivhb bKDjvz"
      >
        <p
          id="content"
        >
          Room2
        </p>
      </div>
      <div
        class="sc-bxivhb iuOKpD"
      >
        <p
          id="content"
        >
          BattleRoyal
        </p>
      </div>
      <div
        class="sc-bxivhb iuOKpD"
      >
        <p
          id="content"
        >
          3/10
        </p>
      </div>
      <div
        class="sc-bxivhb iuOKpD"
      >
        <p
          id="content"
        >
          No
        </p>
      </div>
    </div>
  </div>
</div>
`;
